numpy/ma/tests/test_core.py

```diff
@@ -238,6 +238,26 @@ def test_creation_with_list_of_maskedarrays(self):
         assert_equal(data, [[0, 1, 2, 3, 4], [4, 3, 2, 1, 0]])
         assert_(data.mask is nomask)
 
+    def test_creation_with_list_of_maskedarrays_no_bool_cast(self):
+        # Tests the regression in gh-18551
+        masked_str = np.ma.masked_array(['a', 'b'], mask=[True, False])
+        normal_int = np.arange(2)
+        res = np.ma.asarray([masked_str, normal_int])
```

Comment:
```suggestion
        res = np.ma.asarray([masked_str, normal_int], dtype="U21")
```
Manually setting the dtype, otherwise this might hit the string promotion futurewarning/error